discord/guild.py

```diff
@@ -3836,4 +3836,4 @@ async def create_auto_moderation_rule(
             payload["exempt_channels"] = [c.id for c in exempt_channels]
 
         data = await self._state.http.create_auto_moderation_rule(self.id, payload)
```

Comment:
Move the reason param up here

```suggestion
        data = await self._state.http.create_auto_moderation_rule(self.id, payload, reason=reason)
```